### ISSUE TITLE
CronJobs running at the wrong time...

### ISSUE THREAD

**Issue by scorsi** [hiddeco/cronjobber#34]
Hello,

I have a CronJob set to `0 1 * * *` in `Europe/Paris` which is running at 3 CEST. Another one scheduled at `0 3 * * *` in `Europe/Paris` ran at 5 CEST...

Do you have any idea why it's not running at the time it was scheduled ?

Thanks,

**Comment by u6f6o** [comment]
Hello, 

Same issue here. We scale up our pods in the evening: 

![image](https://user-images.githubusercontent.com/1176370/97002561-6f07ca00-153a-11eb-8c5a-1a9a5d179180.png)

, yet they have been triggered at the wrong time today: 

![image](https://user-images.githubusercontent.com/1176370/97002691-9ced0e80-153a-11eb-89aa-acc313505e39.png)

Best, 
u6f6o

**Comment by pscheid92** [comment]
Hello, @u6f6o and I have the exact same problem.

We have 4 different namespaces with 2 jobs each.
Of those jobs, 4 are scheduled for 19:50 PM and 4 are scheduled for 20:30 PM, each day.

Today, they all ran at exactly the same time (10:44 AM).
Maybe it is a coincidence, but the `updatetz` container logged the right before all jobs ran:

```
2020-10-23T08:43:58+0000 Local Time Zone database updated to version 2020c on /tmp/zoneinfo
```

Our local time zone is CEST (`+0200`), thus the database update was logged at 10:43:58 AM local time.

**Comment by scorsi** [comment]
May be related to #24. Actually don't have the workload to investigate deeper. Switched to default Kubernetes CronJob while the issue is not fixed/investigated.

**Comment by obitech** [comment]
Running into the same problem here. Now my `TZCronJobs` won't start at all, at least not during my specified time. I haven't been able to debug the tzdb inside the container, has there been an issue with the update?

**Comment by hiddeco** [comment]
Sorry about this unexpected behaviour, and I hope it did not result in too much chaos. Can you all please confirm you are using the latest `cronjobber-updatetz` release (`v0.1.1`)? This helps me narrow my search.

**Comment by pscheid92** [comment]
Yes, we (@u6f6o, @obitech and me) used the `v0.1.1` release (with `sidecar.enabled=true`; installed per Helm chart).

**Comment by apicht** [comment]
Just noticed this is happening for us as well. Running `cronjobber:0.2.0` and `cronjobber-updatetz:0.1.0`. Latest log message from updatetz is
```
2020-10-22T23:03:54+0000 Local Time Zone database updated to version 2020c on /tmp/zoneinfo
```

**Comment by maksimu** [comment]
Yeap, same for me. Hopefully it can be fixed soon

**Comment by maksimu** [comment]
When I look at the logs of the cronjobber I see this errors that keep repeating over and over again:

```
2020-10-25T07:09:06.312819787Z cronjobber E1025 07:09:06.312568       1 reflector.go:134] github.com/hiddeco/cronjobber/pkg/client/informers/externalversions/factory.go:117: Failed to list *v1alpha1.TZCronJob: tzcronjobs.cronjobber.hidde.co is forbidden: User "system:serviceaccount:bama-prod:cronjobber" cannot list resource "tzcronjobs" in API group "cronjobber.hidde.co" at the cluster scope
2020-10-25T07:09:07.182842095Z cronjobber E1025 07:09:07.182647       1 controller.go:122] can't list Jobs: jobs.batch is forbidden: User "system:serviceaccount:bama-prod:cronjobber" cannot list resource "jobs" in API group "batch" at the cluster scope
2020-10-25T07:09:07.316420688Z cronjobber E1025 07:09:07.316221       1 reflector.go:134] github.com/hiddeco/cronjobber/pkg/client/informers/externalversions/factory.go:117: Failed to list *v1alpha1.TZCronJob: tzcronjobs.cronjobber.hidde.co is forbidden: User "system:serviceaccount:bama-prod:cronjobber" cannot list resource "tzcronjobs" in API group "cronjobber.hidde.co" at the cluster scope
```

**Comment by obitech** [comment]
@maksimu Your errors _should_  be unrelated. Your service account seems to be missing the correct `ClusterRole` permissions.

```
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  name: cronjobber
rules:
- apiGroups: ['cronjobber.hidde.co']
  resources: ['*']
  verbs: ['*']
- apiGroups: ['batch']
  resources: ['jobs']
  verbs: ['*']
- apiGroups: ['']
  resources: ['events']
  verbs: ['create', 'patch', 'update']
```

**Comment by hiddeco** [comment]
The problem is due to `zic` changing the default format from `fat` to `slim` since `>=2020-b`, this format is however not correctly understood by Go (https://github.com/golang/go/issues/42138), resulting in an UTC fallback.

The best option here is likely to start using Go 1.15, and make use of the embedded timezone information there, which makes the timezone update utility obsolete.

**Comment by obitech** [comment]
@hiddeco do find time to fix this? Otherwise I can draft a PR.

**Comment by hiddeco** [comment]
I have just crafted a release, once it has been pushed to the registry, using the `https://raw.githubusercontent.com/hiddeco/cronjobber/master/deploy/deploy.yaml` manifest with `hiddeco/cronjobber:0.3.0` should solve this issue with a workaround.

I will track the Go issue to release a new version once it is able to understand the `slim` format as well, so the `updatetz` utility can be used again. Keeping this issue open until then.

**Comment by obitech** [comment]
Works again with `0.3.0`, thank you! 🙏

**Comment by hiddeco** [comment]
The bug fix for this problem has landed in the release branch for Go `1.15.4`, with the release scheduled for next week. Expect a patch release as soon as this version is available in container / image land. :sunflower:

**Comment by rmak-cpi** [comment]
I am just wondering when cronjobber is going to be patched with the golang fix (starting from 1.15.4) for handling slim tzdata given that we are already past the 2021 DST transition. Thanks!

**Comment by hiddeco** [comment]
Hello all,

It is likely not what you hoped, but I do at present not have the bandwidth to further maintain this project. See https://github.com/hiddeco/cronjobber/issues/43#issuecomment-890097573 for more information.
